xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -626,19 +629,6 @@ private static String prefixedClusterSpecifierPluginName(String pluginName) {
     return "cluster_specifier_plugin:" + pluginName;
   }
 
-  private static final class FailingConfigSelector extends InternalConfigSelector {
```

Comment:
For my education, would the reason for deleting this `FailingConfigSelector` is that we "generalized" the normal `ConfigSelector`(line 387) to handle this error case (line 398-400)?

Reply:
This part of the description is relevant:

> To give the correct error message, selectConfig() must fail such RPCs directly, and once it can do that there's no need to stop using the config selector in error cases. This has the benefit of fewer moving parts and more consistent threading among cases.

Basically, this FailingConfigSelector doesn't work for all states an RPC may be in. To fix it, we _must_ have the regular ConfigSelector handle the error. And if it is handling the error, we don't need this other config selector.